src/nektar_interface/basis_reference.cpp

```diff
@@ -0,0 +1,461 @@
+#include <nektar_interface/basis_reference.hpp>
+
+namespace NESO::BasisReference {
+
+/**
+ *  Reference implementation to compute eModified_A at an order p and point z.
+ *
+ *  @param p Polynomial order.
+ *  @param z Point in [-1, 1] to evaluate at.
+ *  @returns Basis function evaluated at point.
+ */
+double eval_modA_i(const int p, const double z) {
+  const double b0 = 0.5 * (1.0 - z);
+  const double b1 = 0.5 * (1.0 + z);
+  if (p == 0) {
+    return b0;
+  }
+  if (p == 1) {
+    return b1;
+  }
+  return b0 * b1 * jacobi(p - 2, z, 1, 1);
+}
+
+/**
+ *  Reference implementation to compute eModified_B at an order p,q and point z.
+ *
+ *  @param p First index for basis.
+ *  @param q Second index for basis.
+ *  @param z Point in [-1, 1] to evaluate at.
+ *  @returns Basis function evaluated at point.
+ */
+double eval_modB_ij(const int p, const int q, const double z) {
+
+  double output;
+
+  if (p == 0) {
+    output = eval_modA_i(q, z);
+  } else if (q == 0) {
+    output = std::pow(0.5 * (1.0 - z), (double)p);
```

Comment:
Converting int p to a double in pow will make it slower, unless that's outdated info.

Reply:
done